Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs

```diff
@@ -86,9 +84,8 @@ protected void PrepareTest ([NotNull] GenericPageTestMethodAttribute attribute,
       if (string.IsNullOrWhiteSpace (text))
         throw new InvalidOperationException ("The generic test page did not provide any test information.");
 
-      var serializer = new JavaScriptSerializer();
-      serializer.RegisterConverters (new[] { GenericTestParameterConverter.Instance });
-      var dto = serializer.Deserialize<GenericTestPageParameterDto> (text);
+      var serializerOptions = new JsonSerializerOptions { Converters = { GenericTestParameterConverter.Instance } };
+      var dto = JsonSerializer.Deserialize<GenericTestPageParameterDto> (text, serializerOptions);
```

Comment:
The documentation for exceptions thrown for the `Deserialize<T>` method differ a bit between `System.Text.Json` and the old `JavaScriptSerializer` ([Js](https://docs.microsoft.com/en-us/dotnet/api/system.web.script.serialization.javascriptserializer.deserialize?view=netframework-4.8#System_Web_Script_Serialization_JavaScriptSerializer_Deserialize__1_System_String_), [Text.Json](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-5.0#System_Text_Json_JsonSerializer_Deserialize__1_System_String_System_Text_Json_JsonSerializerOptions_)).

The reasons for throwing an exception will probably boil down to being approximately the same (maybe except the things with `__type`?). But just to be sure, since it's only for internal usage, I guess it's okay to have different exception types thrown in case somethings goes bad?

Reply:
I do think so, I'll leave this open for the lead review.

Reply:
@MichaelKetting we do not want any exception, still there is no graceful way of dealing with an error at this point, as no sensible fallback mechanism can be provided that wouldn't hide the original error making it even harder to fix it. But that's just my opinion, I'm open for suggestions.

Reply:
@iam-mholle okay, I'm confused. I figure it's an exception with the old code and an exception with the new code?

Reply:
It is.

I guess I misunderstood the "dealing with the exception" part in the following.

> since this is just our test code, we a) never want to haven an expcetion thrown anyway and b) **will deal with the exceptions if and when they happen**

I thought you mean exception handling in the source code, not handling the error by fixing it when it causes failing tests.